lua/go/health.lua

```diff
@@ -139,11 +138,21 @@ local function path_check()
   return true
 end
 
+local function goenv()
+  local env = {}
+  local raw = vim.fn.system('go env')
+  for key, value in string.gmatch(raw, "([^=]+)='([^']*)'\n") do
+    env[key] = value
+  end
+  return env
+end
+
 local function env_check()
-  local envs = { 'GOROOT', 'GOBIN' }
+  local env = goenv()
+  local keys = { 'GOROOT', 'GOBIN' }
   local any_warn = false
-  for _, env in ipairs(envs) do
-    if vim.env[env] == nil then
+  for _, key in ipairs(keys) do
+    if env[key] == nil then
```

Comment:
the go env will output something like `GOEXE=''` so in those case I would say the env is also not set. Or, you could change the gmatch on line 144 so only the non-empty value will be added to env.

Reply:
Will adjust. You meant values should be `nil` instead of `''`. Right?